fem/lor/lor_batched.cpp

```diff
@@ -48,12 +48,18 @@ bool HasIntegrators(BilinearForm &a)
 
 bool BatchedLORAssembly::FormIsSupported(BilinearForm &a)
 {
-   const FiniteElementCollection *fec = a.FESpace()->FEColl();
+   const FiniteElementSpace *fes = a.FESpace();
+   const FiniteElementCollection *fec = fes->FEColl();
    // TODO: check for maximum supported orders
 
    // Batched LOR requires all tensor elements
    if (!UsesTensorBasis(*a.FESpace())) { return false; }
 
+   // Batched LOR requires all elements to be similar
+   const auto *mesh = fes->GetMesh();
+   if (fes->IsVariableOrder() || mesh->Nonconforming() ||
+       mesh->GetNumGeometries(mesh->Dimension()) > 1) { return false; }
```

Comment:
It should work with nonconforming meshes (in serial, there was a bug in this case, fixed in #4891). I haven't tested with variable orders (in principle it could be made to work using the $P$ matrix).